api/src/main/java/io/minio/MinioClient.java

```diff
@@ -2383,6 +2386,226 @@ public void copyObject(String bucketName, String objectName, Map<String,String>
+      throw new InvalidArgumentException("compose sources cannot be empty");
+    }
+
+    if (sse != null && !this.baseUrl.isHttps()) {
+      if (sse.type() == ServerSideEncryption.Type.SSE_C) {
+        throw new InvalidArgumentException(
+          "SSE_C operations must be performed over a secure connection.");
+      } else if (sse.type() == ServerSideEncryption.Type.SSE_KMS ) {
+        throw new InvalidArgumentException("SSE_KMS operations must be performed over a secure connection.");
+      }
+    }
+
+    long objectSize = 0;
+    int partsCount = 0;
+    for (int i = 0; i < sources.size(); i++) {
+      ComposeSource src = sources.get(i);
+
+      ObjectStat stat = statObject(src.bucketName(), src.objectName(), src.sse());
+      src.buildHeaders(stat.length(), stat.etag());
+
+      if (i != 0 && src.headers().containsKey("x-amz-meta-x-amz-key")) {
+        throw new InvalidArgumentException("Client side encryption is not supported for more than one source");
+      }
+
+      long size = stat.length();
+      if (src.length() != null) {
+        size = src.length();
+      }
+      if (src.offset() != null) {
+        size -= src.offset();
+      }
+
+      if (size < MIN_MULTIPART_SIZE && sources.size() != 1 && i != (sources.size() - 1)) {
+        throw new InvalidArgumentException("source " + src.bucketName() + "/" + src.objectName() + ": size "
+          + size + " must be greater than " + MIN_MULTIPART_SIZE);
+      }
+
+      objectSize += size;
+      if (objectSize > MAX_OBJECT_SIZE) {
+        throw new InvalidArgumentException("Destination object size must be less than " + MAX_OBJECT_SIZE);
+      }
+
+      if (size > MAX_PART_SIZE) {
+        long count = size / MAX_PART_SIZE;
+        long lastPartSize = size - (count * MAX_PART_SIZE);
+        if (lastPartSize > 0) {
+          count++;
+        } else {
+          lastPartSize = MAX_PART_SIZE;
+        }
+
+        if (lastPartSize < MIN_MULTIPART_SIZE && sources.size() != 1 && i != (sources.size() - 1)) {
+          throw new InvalidArgumentException("source " + src.bucketName() + "/" + src.objectName() + ": "
+            + "for multipart split upload of " + size
+            + ", last part size is less than " + MIN_MULTIPART_SIZE);
+        }
+
+        partsCount += (int)count;
+      } else {
+        partsCount++;
+      }
+
+      if (partsCount > MAX_MULTIPART_COUNT) {
+        throw new InvalidArgumentException("Compose sources create more than allowed multipart count "
+          + MAX_MULTIPART_COUNT);
+      }
+    }
+
+    if (partsCount == 1) {
+      ComposeSource src = sources.get(0);
+      copyObject(bucketName, objectName, headerMap, sse, src.bucketName(), src.objectName(), src.sse(),
+          src.copyConditions());
+      return;
+    }
+    if (headerMap == null) {
+      headerMap = new HashMap<>();
+    }
+
+    if (sse != null) {
+
+//
+//      sse.marshal(headerMap);
+      checkWriteRequestSse(sse);
+      headerMap.putAll(sse.headers());
```

Comment:
No need to  `headerMap = new HashMap<>();`, you could do 
```java
if (sse != null) {
   headerMap = sse.headers();
}
```

Reply:
In case sse is null  `headerMap` needs to be initialled as `initMultipartUpload` gets the Content-type  `headerMap.get("Content-Type") == null` . If the map is not initialised it causes a null pointer exception.

Reply:
As per https://docs.aws.amazon.com/AmazonS3/latest/API/mpUploadInitiate.html, `Content-Type` is an optional header. The check can be removed in `initMultipartUpload()` or you could deal with `null` `headerMap` in `initMultipartUpload()` itself which is clean.

Reply:
`headerMap = new HashMap<>();` is needed now as `uploadPartCopy` takes headerMap which contains encryption headers for source as well as destination.  In order to populate `headerMap` with source `headers` it has been initialised.